widget/widgetAssets/js/socOverviewSankey.service.js

```diff
@@ -110,42 +110,57 @@
-                  'field': currentLayer['targetNodeType'] === 'picklist' || currentLayer['targetNodeType'] === 'manyToMany' ? currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.itemValue' : currentLayer['targetNodeSubField']
-              });
-              if (currentLayer['targetNodeType'] === 'picklist' || currentLayer['targetNodeType'] === 'manyToMany') {
-                  queryObject.aggregates.push({
-                      'operator': 'groupby',
-                      'alias': 'series_' + elementIndex + '_color',
-                      'field': currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.color'
-                  });
-              }
-          }
-  
-  
-          function getFilters(duration) {
-              let frontFilter = {};
-              frontFilter.logic = 'AND';
-              if (config.entityTrackable) {
-                  frontFilter.filters = [{
-                      field: 'createDate',
-                      operator: 'gte',
-                      value: currentDateMinusService(duration),
-                      type: 'datetime'
-                  }];
-              }
-  
-              let dataFilters = config.filters ? angular.copy(config.filters) : {};
-              if (dataFilters.filters) {
-                  dataFilters.filters.push(frontFilter);
-              } else {
-                  dataFilters = frontFilter;
-              }
-  
-              return dataFilters;
-          }
+            queryObject.aggregates.push({
+                'operator': 'groupby',
+                'alias': 'series_' + elementIndex,
+                'field': currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany' ? currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.itemValue' : currentLayer['targetNodeSubField']
+            });
+            if (currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany') {
+                queryObject.aggregates.push({
+                    'operator': 'groupby',
+                    'alias': 'series_' + elementIndex + '_color',
+                    'field': currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.color'
+                });
+            }
+        }
+
+
+        function getFilters(duration) {
+            let frontFilter = {};
+            frontFilter.filters = [];
+            for (var i = 0; i < config.layers.length; i++) {
+                let currentLayer = config.layers[i];
+                let _nullField = '';
+                if(currentLayer['targetNodeSubField'] !== ''){
+                    _nullField = currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany' ? currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.itemValue' : currentLayer['targetNodeSubField'];
```

Comment:
Same here in the else condition of the ternary operator it should be currentLayer['targetNodeField'].currentLayer['targetNodeSubField']